packages/use-segment/src/useSegment.tsx

```diff
@@ -52,34 +54,49 @@ function SegmentProvider<T extends Events>({
   children,
   settings,
   initOptions,
+  areOptionsLoading = false,
   onError,
   onEventError,
   events,
 }: SegmentProviderProps<T>) {
+  const [isAnalyticsReady, setIsAnalyticsReady] = useState(false)
   const [internalAnalytics, setAnalytics] = useState<Analytics | undefined>(
     undefined,
   )
 
   const shouldLoad = useMemo(() => {
+    // If options are loading (for example Segment integrations),
+    // we don't know if we should load or not
+    if (areOptionsLoading) {
+      return undefined
+    }
+
     const hasNoIntegrationsSettings = !initOptions?.integrations
-    const isAllEnabled = !!initOptions?.integrations?.All
     const isAnyIntegrationEnabled = Object.values(
       initOptions?.integrations ?? {},
     ).reduce<boolean>((acc, integration) => !!acc || !!integration, false)
 
     return (
       !!settings?.writeKey &&
-      (hasNoIntegrationsSettings || isAllEnabled || isAnyIntegrationEnabled)
+      (hasNoIntegrationsSettings || isAnyIntegrationEnabled)
     )
-  }, [initOptions?.integrations, settings?.writeKey])
+  }, [areOptionsLoading, initOptions, settings])
```

Comment:
```suggestion
  }, [areOptionsLoading, initOptions?.integrations, settings?.writeKey])
```

Reply:
This will re-render everytime the object change not what you trutly use ?

Reply:
Good catch, my IDE is somehow too optimistic ;)